src/framework/theme/components/select/select.component.ts

```diff
@@ -1168,10 +1173,10 @@ export class NbSelectComponent implements OnChanges, AfterViewInit, AfterContent
     const previouslySelectedOptions = this.selectionModel;
     this.selectionModel = [];
 
-    if (isArray) {
-      value.forEach(option => this.selectValue(option));
+    if (this.multiple) {
+      formattedValue.forEach(option => this.selectValue(option));
     } else {
-      this.selectValue(value);
+      this.selectValue(formattedValue);
```

Comment:
We also need to update `selectValue` method. As per documentation we treat `null` and `undefined` value as reset value, so no option should become selected when value becomes `null` or `undefined`.